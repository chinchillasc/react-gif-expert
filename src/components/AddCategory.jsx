import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {
    
    const [inputValue, setInputValue] = useState('');

    const onImputChange = ({target}) => {
        setInputValue(target.value);
    }
    
    const onSubmit = (event) => { 
        event.preventDefault();
        onNewCategory(inputValue);   
        setInputValue('');        
    }

    return (
        <form onSubmit={onSubmit}>
            <input 
                type="text" 
                placeholder="Buscar gif"
                value={inputValue}
                onChange={onImputChange}
            />
        </form>
    )
}