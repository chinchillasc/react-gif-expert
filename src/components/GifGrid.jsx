import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { PropTypes } from 'prop-types'

export const GifGrid = ({categoria}) => {

    const {imagenes,isLoading} = useFetchGifs(categoria);

    return (
        <>
            <h3>{categoria}</h3>

            { isLoading && (<h2>Cargando...</h2>) }
            
            <div className="card-grid">
                {
                    imagenes.map((imagen) => (
                        <GifGridItem 
                            key={imagen.id} 
                            {...imagen}
                        />
                    ))
                }
            </div>    
        </>
    );
}

GifGrid.propType = {
    categoria: PropTypes.string.isRequired
}