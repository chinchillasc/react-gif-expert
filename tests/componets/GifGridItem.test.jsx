import { render,screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Prueba en <GifGridItem/>', () => {

    const title = 'Goku';
    const url = 'https://dragon-ball.com/goku.gif';

    test('Debe hacer match con el snapshot', () => { 
        const {container} = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
     });

     test('Debe mostrar la imagen con el url y el Alt indicado', () => { 
        render(<GifGridItem title={title} url={url} />);
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
     });
    
     test('Debe mostrar el titulo en nuestro componente', () => { 
        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
     });
});