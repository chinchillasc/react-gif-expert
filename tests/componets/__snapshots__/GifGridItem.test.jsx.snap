// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://dragon-ball.com/goku.gif"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
